src/nvim/eval.c

```diff
@@ -17685,6 +17685,15 @@ static void f_writefile(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     EMSG2(_(e_listarg), "writefile()");
     return;
   }
+  const list_T *const list = argvars[0].vval.v_list;
+  if (list == NULL) {
+    return;
+  }
```

Comment:
This change seems to contradict the existing test `NULL list is accepted as an empty list by writefile()`. 
I think we can leave this out, because we don't consider it an error. cc @ZyX-I

Reply:
NULL list should still create a file, it is incorrect to exit here.